src/compiler/llvm-to-backend/host/LLVMToHost.cpp

```diff
@@ -91,57 +88,71 @@ bool LLVMToHostTranslator::toBackendFlavor(llvm::Module &M, PassHandler &PH) {
     return false;
 
   llvm::ModulePassManager MPM;
-  PH.ModuleAnalysisManager->clear(); // for some reason we need to reset the analyses... otherwise
-                                     // we get a crash at IPSCCP
+  // TODO: Clean me up.
+  // PH.ModuleAnalysisManager->clear(); // for some reason we need to reset the analyses... otherwise
+  //                                    // we get a crash at IPSCCP
+
+  llvm::LoopAnalysisManager LAM;
+  llvm::FunctionAnalysisManager FAM;
+  llvm::CGSCCAnalysisManager CGAM;
+  llvm::ModuleAnalysisManager MAM;
 
   PH.PassBuilder->registerAnalysisRegistrationCallback([](llvm::ModuleAnalysisManager &MAM) {
     MAM.registerPass([] { return SplitterAnnotationAnalysis{}; });
   });
-  PH.PassBuilder->registerModuleAnalyses(*PH.ModuleAnalysisManager);
+  // PH.PassBuilder->registerModuleAnalyses(*PH.ModuleAnalysisManager);
+
+  PH.PassBuilder->registerModuleAnalyses(MAM);
+  PH.PassBuilder->registerCGSCCAnalyses(CGAM);
+  PH.PassBuilder->registerFunctionAnalyses(FAM);
+  PH.PassBuilder->registerLoopAnalyses(LAM);
+  PH.PassBuilder->crossRegisterProxies(LAM, FAM, CGAM, MAM);
   registerCBSPipeline(MPM, hipsycl::compiler::OptLevel::O3, true);
+  HIPSYCL_DEBUG_INFO << "LLVMToHostTranslator: Done registering\n";
 
   llvm::FunctionPassManager FPM;
   FPM.addPass(HostKernelWrapperPass{KnownLocalMemSize});
   MPM.addPass(llvm::createModuleToFunctionPassAdaptor(std::move(FPM)));
 
-  MPM.run(M, *PH.ModuleAnalysisManager);
-
+  // MPM.printPipeline(llvm::outs(),[](llvm::StringRef Name){return Name;});
+  MPM.run(M, MAM);
+  HIPSYCL_DEBUG_INFO << "LLVMToHostTranslator: Done toBackendFlavor\n";
   return true;
 }
 
 bool LLVMToHostTranslator::translateToBackendFormat(llvm::Module &FlavoredModule,
                                                     std::string &out) {
-  auto InputFile = llvm::sys::fs::TempFile::create("acpp-sscp-host-%%%%%%.bc");
-  auto OutputFile = llvm::sys::fs::TempFile::create("acpp-sscp-host-%%%%%%.so");
 
-  if (auto E = InputFile.takeError()) {
-    this->registerError("LLVMToHost: Could not create temp file: " + InputFile->TmpName);
+  llvm::SmallVector<char> InputFile;
+  int InputFD;
+  // don't use fs::TempFile, as we can't unlock the file for the clang invocation later... (Windows)
+  if(auto E = llvm::sys::fs::createTemporaryFile("acpp-sscp-host", "bc", InputFD, InputFile, llvm::sys::fs::OF_None)){
```

Comment:
If there is a problem with locking on Windows, can we do this such that we still use the properly file-locked code path on Linux?